File: src/main/java/net/adventuregame/items/Item.java
package net.adventuregame.items;

import net.adventuregame.entities.Entity;
import net.adventuregame.items.entities.ItemEntity;
import net.adventuregame.models.TexturedModel;
import org.joml.Vector3f;

public abstract class Item{
    private String name;

    public Item(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public abstract ItemEntity createEntity(Vector3f position);

    public abstract TexturedModel getGuiModel();
}

File: src/main/kotlin/net/adventuregame/items/Item.kt
package net.adventuregame.items

import net.adventuregame.items.entities.ItemEntity
import net.adventuregame.models.TexturedModel
import org.joml.Vector3f

abstract class Item(val name: String?) {
    abstract fun createEntity(position: Vector3f): ItemEntity?

    abstract fun getGuiModel(): TexturedModel?
}
